libsel4muslcsys/src/force_init.c

```diff
@@ -0,0 +1,16 @@
+/*
+ * Copyright 2020, Data61, CSIRO (ABN 41 687 119 230)
+ *
+ * SPDX-License-Identifier: GPL-2.0-only
+ */
+
+/*
+ * This file is linked into every direct dependency of muslcsys and will
+ * for the linker to include the init function and add it as a
+ * constructor.
+ *
+ * The constructor sill only exists as an object in muslcsys so that it
+ * will only ever be linked in once.
+ */
+
+extern void muslcsys_init_muslc(void);
```

Comment:
I don't think this will lead to the symbol becoming linked in as it isn't going to create an undefined symbol in the compiled object file.

Reply:
Fixed. I just put the constructor in `vsyscall.c` and added an `INTERFACE` link option to force `muslcsys_init_muslc` to be an undefined symbol. I should have done it that way to begin with.